doc/develop/west/build-flash-debug.rst

```diff
@@ -476,6 +562,23 @@ For example, to print usage information about the ``jlink`` runner::
 
   west flash -H -r jlink
 
+.. _west-multi-domain-flashing:
+
+multi-domain flashing
+=====================
+
+When a :ref:`west-multi-domain-builds` folder is detected, then ``west flash``
+will flash all domains in the order they appear in the  :ref:`west-domains-file`.
+
+It is possible to build a single domain in a multi-domain project by using
+``--domain``.
+
+For example, in a multi-domain build with :ref:`hello_world` and
+`MCUboot`_, you can use the ``--domain hello_world`` domain to only flash
+this domain::
+
+  west flash --domain hello_world
```

Comment:
Some issues related to argument passing here.

sysbuild/index.rst says:

> When invoking ``west flash`` on a build consisting of multiple images, each
> image is flashed in sequence. Extra arguments such as ``--runner jlink`` are
> passed to each invocation.
>
> For more details, see :ref:`west-multi-domain-flashing`.

But the examples here are all about the `--domain` argument, not the `--runner` argument. I think that's confusing; the examples should be consistent so the reader knows they've found the details here.

Second problem: we discussed that some options, should not be handled agnostically by sysbuild-aware flash, such as things like --dev-id which are not implemented by all runners. I don't see documentation here reflecting the decisions you made about how those are handled.

Reply:
> But the examples here are all about the --domain argument, not the --runner argument.

I agree. Leftover from where I had most of the `west build|flash|debug` documentation inside sysbuild, before moving it to `west` page and decided to only have a brief mentioning of this together with a link to `west` docs.
Will clean up.

> Second problem: we discussed that some options, like --dev-id, should not be handled agnostically by sysbuild-aware flash.

I remember we briefly touched upon this but only decided that we be agnostic to all args, and just pass them on.
If a user wants to have, let's say `--dev-id` used for only a given domain, he can do so by also specifying such domain, like this: `west flash --dev-id <id> --domain <dom>`

Reply:
> I remember we briefly touched upon this but only decided that we be agnostic to all args, and just pass them on.

That's odd, I remember the opposite. I thought you were going to have some sort of allowlist of arguments that you would allow that were guaranteed to be domain independent. Then you would fail the command if something outside the allowlist, and therefore potentially domain-specific, was given.

Reply:
I made a PR here: https://github.com/zephyrproject-rtos/zephyr/pull/48914

This will print a warning to the user if he does:
```
$ west flash --<runner-arg>
```
like
```
$ west flash --dev-id 12345
```
on a multi domain build.

But if user does:
```
$ west flash --dev-id 12345 --domain <domain>
```
no warning is printed.

The allows users to still apply runners args and let the user decide if it makes sense, while at the same time he is warned that that currently this feature is considered experimental.

Allows us to get some feedback on this feature.
Should we open a common issue that we could link to, so we collect experiences at a single place, like:
```
> WARNING: Specifying runner options for multiple domains is experimental.
> If problems are experienced, please specify a single domain using '--domain <domain>'         
> Any feedback on this feature can be given https://github.com/zephyrproject-rtos/zephyr/issues/<a_number>
```